lib/os/ring_buffer.c

```diff
@@ -36,56 +36,48 @@ static uint32_t mod(struct ring_buf *buf, uint32_t val)
 	return likely(buf->mask) ? val & buf->mask : val % buf->size;
 }
 
-/* Check if indexes did not progress too far (too close to 32-bit wrapping).
- * If so, then reduce all indexes by an arbitrary value.
- */
-static void item_indexes_rewind(struct ring_buf *buf)
+static uint32_t get_rewind_value(uint32_t buf_size, uint32_t threshold)
 {
-	uint32_t rew;
-	uint32_t threshold = ring_buf_get_rewind_threshold();
-
-	if (buf->head < threshold) {
-		return;
-	}
+	return buf_size * (threshold / buf_size);
+}
 
-	rew = buf->size * (threshold / buf->size);
+int ring_buf_is_empty(struct ring_buf *buf)
+{
+	uint32_t tail = buf->tail;
+	uint32_t head = buf->head;
 
-	k_spinlock_key_t key = k_spin_lock(&buf->lock);
+	if (0 && tail < head) {
```

Comment:
This condition will always be false.

Reply:
eh, was testing something and forgot to clear it.